rest_framework-stubs/permissions.pyi

```diff
@@ -44,7 +44,11 @@ class NOT(_SupportsHasPermission):
     def __init__(self, op1: _SupportsHasPermission) -> None: ...
 
 class BasePermissionMetaclass(OperationHolderMixin, type): ...  # type: ignore[misc]
-class BasePermission(_SupportsHasPermission, metaclass=BasePermissionMetaclass): ...  # type: ignore[misc]
+
+class BasePermission(metaclass=BasePermissionMetaclass):  # type: ignore[misc]
+    def has_permission(self, request: Request, view: APIView) -> bool: ...
+    def has_object_permission(self, request: Request, view: APIView, obj: Any) -> bool: ...
```

Comment:
If I kept `_SupportsHasPermission` here, it causes "Metaclass conflict" errors.

I don't understand metaclasses well enough to say whether that's a bug in mypy or a genuine issue.

But duplicating these method signatures here fixes it.